homeassistant/components/control4/config_flow.py

```diff
@@ -90,80 +79,81 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
-                controller_unique_id = hub.controller_unique_id
                 mac = (controller_unique_id.split("_", 3))[2]
                 formatted_mac = format_mac(mac)
+                data = {
+                    CONF_HOST: user_input[CONF_HOST],
+                    CONF_USERNAME: user_input[CONF_USERNAME],
+                    CONF_PASSWORD: user_input[CONF_PASSWORD],
+                    CONF_CONTROLLER_UNIQUE_ID: controller_unique_id,
+                }
                 await self.async_set_unique_id(formatted_mac)
                 self._abort_if_unique_id_configured()
                 return self.async_create_entry(
                     title=controller_unique_id,
-                    data={
-                        CONF_HOST: user_input[CONF_HOST],
-                        CONF_USERNAME: user_input[CONF_USERNAME],
-                        CONF_PASSWORD: user_input[CONF_PASSWORD],
-                        CONF_CONTROLLER_UNIQUE_ID: controller_unique_id,
-                    },
+                    data=data,
                 )
 
         return self.async_show_form(
             step_id="user", data_schema=DATA_SCHEMA, errors=errors
         )
 
-    @staticmethod
-    @callback
-    def async_get_options_flow(config_entry):
-        """Get the options flow for this handler."""
-        return OptionsFlowHandler(config_entry)
-
-
-class OptionsFlowHandler(config_entries.OptionsFlow):
-    """Handle a option flow for Control4."""
-
-    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
-        """Initialize options flow."""
-        self.config_entry = config_entry
-
-    async def async_step_init(self, user_input=None):
-        """Handle options flow."""
+    async def async_step_user_reauth(self, user_input=None):
+        """Handle a reauthentication request."""
+        errors = {}
         if user_input is not None:
-            return self.async_create_entry(title="", data=user_input)
-
-        data_schema = vol.Schema(
-            {
-                vol.Optional(
-                    CONF_SCAN_INTERVAL,
-                    default=self.config_entry.options.get(
-                        CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
-                    ),
-                ): vol.All(cv.positive_int, vol.Clamp(min=MIN_SCAN_INTERVAL)),
-            }
+            errors, controller_unique_id = await self._validate_input(user_input)
+            if not errors:
+                mac = (controller_unique_id.split("_", 3))[2]
+                formatted_mac = format_mac(mac)
+                data = {
+                    CONF_HOST: user_input[CONF_HOST],
+                    CONF_USERNAME: user_input[CONF_USERNAME],
+                    CONF_PASSWORD: user_input[CONF_PASSWORD],
+                    CONF_CONTROLLER_UNIQUE_ID: controller_unique_id,
+                }
+                _LOGGER.debug("Reauthentication occurring")
+                existing_entry = await self.async_set_unique_id(formatted_mac)
```

Comment:
We need to validate `existing_entry` isn't None and abort in case they enter credentials for another location